packages/manager/src/views/WebPropertyDetailPage/components/SSR/SsrForm.tsx

```diff
@@ -0,0 +1,282 @@
+import { toast } from 'react-hot-toast';
+import * as yup from 'yup';
+
+type FormData = {
+  propertyIdentifier: string;
+  name: string;
+  path: string;
+  ref: string;
+  env: string;
+  identifier: string;
+  nextRef: string;
+  accessUrl: string;
+  updatedAt: string;
+  _id: number;
+  isSSR: boolean;
+  healthCheckPath: string;
+  config: {
+    [key: string]: string;
+  };
+  imageUrl: string;
+};
+
+const schema = yup.object().shape({
+  name: yup.string().required(),
+  path: yup.string().required(),
+  imageUrl: yup.string().required(),
+  healthCheckPath: yup.string().required()
+});
+
+type Props = {
+  onClose: () => void;
+  propertyIdentifier: string;
+};
+
+export const SSRForm = ({ onClose, propertyIdentifier }: Props): JSX.Element => {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<FormData>({
+    resolver: yupResolver(schema)
+  });
+
+  const createSsrSpaProperty = useAddSsrSpaProperty(propertyIdentifier);
+  const [keyValuePairs, setKeyValuePairs] = useState([{ key: '', value: '' }]);
+  const handleAddKeyValuePair = () => {
+    setKeyValuePairs([...keyValuePairs, { key: '', value: '' }]);
+  };
+
+  const handleKeyValuePairChange = (index: number, key: string, value: string) => {
+    const newKeyValuePairs = [...keyValuePairs];
+    newKeyValuePairs[index] = { key, value };
+    setKeyValuePairs(newKeyValuePairs);
+  };
+
+  const webProperties = useGetWebPropertyGroupedByEnv(propertyIdentifier);
+
+  const webPropertiesKeys = Object.keys(webProperties.data || {});
+
+  const onSubmit = async (dataf: FormData) => {
+    const json: Record<string, string> = {};
+    keyValuePairs.forEach(({ key, value }) => {
+      if (key && value) {
+        json[key] = value;
+      }
+    });
+    const temp = dataf;
+    if (!temp.path.startsWith('/')) {
+      temp.path = `/${temp.path}`;
+    }
+    if (!temp.healthCheckPath.startsWith('/')) {
+      temp.healthCheckPath = `/${temp.healthCheckPath}`;
+    }
+
+    const newData = {
+      ...temp,
+      config: json,
+      propertyIdentifier
+    };
+    try {
+      await createSsrSpaProperty.mutateAsync(newData);
+      onClose();
+      toast.success('Deployed SSR successfully');
+    } catch (error) {
+      toast.error('Failed to deploy SSR');
+    }
+  };
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      <Split hasGutter>
+        <SplitItem isFilled>
+          <FormGroup label="Name" isRequired>
+            <Controller
+              name="name"
+              control={control}
+              render={({ field }) => (
+                <TextInput
+                  id="name"
+                  type="text"
+                  isRequired
+                  value={field.value}
+                  onChange={field.onChange}
+                  onBlur={field.onBlur}
+                />
+              )}
+            />
+            {errors.name && <span>{errors.name.message}</span>}
```

Comment:
Same suggestion as prev for error messages

Reply:
All the prev suggestion applicable here

Reply:
done